### PR TITLE
Set node attribute for OpenStack-role pacemaker attribute for controller

### PR DESCRIPTION
**Pull request by vuntz**
Depends on https://github.com/crowbar/crowbar-ha/pull/62

### CODE REVIEW

File: crowbar_framework/app/models/neutron_service.rb [hunk 1]
```diff
@@ -380,7 +380,9 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     end
 
     server_elements, server_nodes, server_ha_enabled = role_expand_elements(role, "neutron-server")
+    OpenStack::HA.set_controller_role(server_nodes) if server_ha_enabled
     network_elements, network_nodes, network_ha_enabled = role_expand_elements(role, "neutron-network")
+    OpenStack::HA.set_controller_role(network_nodes) if network_ha_enabled
```
Comment (jsuchome) [inline crowbar_framework/app/models/neutron_service.rb]: What about calling `set_controller_role` only once here, with the union of both sets as argument?

**Comment by jsuchome** [comment]
Not tested, but the code looks correct. 
+1 (with one nitpick)

File: crowbar_framework/app/models/neutron_service.rb (hunk 1 above)
Comment (vuntz) [inline crowbar_framework/app/models/neutron_service.rb]: The `if` conditions are different, so we can't do that.

Comment (jsuchome) [inline crowbar_framework/app/models/neutron_service.rb]: Ouch, silly me :-)

**Comment by vuntz** [comment]
Had to change OpenStack to Openstack to make rails happy :/

**Comment by aplanas** [comment]
+1

**Comment by jsuchome** [comment]
Still +1

**Pull request merged**